code/modules/mob/living/simple_animal/hostile/gym_rats.dm

```diff
@@ -0,0 +1,703 @@
+			else
+				target_turf = get_ranged_target_turf(T, dir, punch_throw_range)
+			M.throw_at(target_turf,100,punch_throw_speed)
+
+/mob/living/simple_animal/hostile/retaliate/roid_rat/UnarmedAttack(var/atom/A)
+	if(is_type_in_list(A, edibles)) // If we click on something edible, it's time to chow down!
+		delayNextAttack(10)
+		chowdown(A)
+	if(is_type_in_list(A, gym_equipments)) // If we click on gym equipment, it's time to work out!
+		A.attack_hand(src, 0, A.Adjacent(src))
+	else return ..()
+
+/mob/living/simple_animal/hostile/retaliate/roid_rat/proc/chowdown(var/atom/eat_this)
+	if(istype(eat_this,/obj/item/weapon/reagent_containers/food/snacks/cheesewedge)) //Mmm, cheese wedge. Gives back a small amount of health upon consumption
+		health+=10
+		visible_message("\The [name] gobbles up \the [eat_this].", "<span class='notice'>You gobble up the [eat_this].</span>")
+		playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
+		flick(icon_eat, src)
+		qdel(eat_this)
+	if(istype(eat_this,/obj/item/weapon/reagent_containers/food/snacks/meat)) //Protein! Gives back a smaller amount of health, but also packs on some extra max hp
+		health+=3
+		maxHealth+=3
+		visible_message("\The [name] gobbles up \the [eat_this].", "<span class='notice'>You gobble up the [eat_this].</span>")
+		playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
+		flick(icon_eat, src)
+		qdel(eat_this)
+	if(istype(eat_this,/obj/item/weapon/reagent_containers/food/snacks/sliceable/cheesewheel)) //A cheese wheel feast! Gives back a lot more health than just a slice
+		health+=50
+		visible_message("\The [name] gobbles up \the [eat_this].", "<span class='notice'>You gobble up the [eat_this].</span>")
+		playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
+		flick(icon_eat, src)
+		qdel(eat_this)
+
+/mob/living/simple_animal/hostile/retaliate/roid_rat/Life() // Copied from hammy wheel running code
+	if(timestopped)
+		return 0
+	. = ..()
+	if(.)
+		if(enemies.len && prob(1)) // Roid rats take much longer to calm down compared to gym rats. Roid rage!
+			Calm()
+	if(maxHealth < 200)
+		melee_damage_lower = 10
+		melee_damage_upper = 20
+		environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | OPEN_DOOR_STRONG
+	if(maxHealth >= 200)
+		melee_damage_lower = 20
+		melee_damage_upper = 30
+		environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | SMASH_WALLS | OPEN_DOOR_STRONG // Maximized gainz will grant roid rats incredible punching power, and the ability to smash through normal walls! Oh YEAAAAAAAAAAAAAAH
+	if(!my_wheel && isturf(loc) && !client) // Roid rats with players in them won't be force moved
+		for(var/obj/O in view(2, src))
+			if(is_type_in_list(O, gym_equipments))
+				my_wheel = locate(O) in view(2, src)
+				wander = FALSE
+				roidratwheel(20)
+				break
+			else
+				wander = TRUE
+				speed = 2
+
+/mob/living/simple_animal/hostile/retaliate/roid_rat/proc/roidratwheel(var/repeat)
```

Comment:
refactor to inherit all this from gym rat

Reply:
Will work on trying to figure this out after getting the roid rat punch spell to work.

Reply:
This should be addressed now.